src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -74,9 +79,8 @@ public BambooService(GitService gitService, ProgrammingSubmissionRepository prog
             Optional<ContinuousIntegrationUpdateService> continuousIntegrationUpdateService, BambooBuildPlanService bambooBuildPlanService, FeedbackRepository feedbackRepository,
             @Qualifier("bambooRestTemplate") RestTemplate restTemplate, @Qualifier("shortTimeoutBambooRestTemplate") RestTemplate shortTimeoutRestTemplate, ObjectMapper mapper,
             UrlService urlService, BuildLogEntryService buildLogService, ExerciseDateService exerciseDateService) {
-        super(programmingSubmissionRepository, feedbackRepository, buildLogService, restTemplate, shortTimeoutRestTemplate);
+        super(programmingSubmissionRepository, feedbackRepository, buildLogService, restTemplate, shortTimeoutRestTemplate, versionControlService);
```

Comment:
I don't like the dependency here to the `versionControlService`. We explicitly avoided this previously to prevent a high coupling. Can you try to find a different solution?